src/components/app/data/utils.js

```diff
@@ -387,18 +384,21 @@ export const transformCourseEnrollment = (rawCourseEnrollment) => {
 };
 
 /**
- * TODO
+ * Groups course enrollments by their status.
  * @param {*} courseEnrollments
  * @returns
  */
 export const groupCourseEnrollmentsByStatus = (courseEnrollments) => {
   const courseEnrollmentsByStatus = Object.keys(COURSE_STATUSES).reduce((acc, status) => {
-    acc[status] = courseEnrollments ? courseEnrollments.filter(
-      courseEnrollment => courseEnrollment.courseRunStatus === COURSE_STATUSES[status],
-    ) : [];
+    // Skip assigned and requested course enrollment statuses; this function is intended to only
+    // group by realized enrollment statuses, not requested/assigned statuses.
+    if (![COURSE_STATUSES.assigned, COURSE_STATUSES.requested].includes(status)) {
+      acc[status] = courseEnrollments ? courseEnrollments.filter(
+        courseEnrollment => courseEnrollment.courseRunStatus === COURSE_STATUSES[status],
+      ) : [];
```

Comment:
[nit/non-blocking] Some missing coverage on the empty state logic.

Reply:
Resolved in 6e766551afad421b06f4b4caa92817757f1aa517

Reply:
That said, I don't think the function is actually ever called without an array passed 🤷‍♂️